### PR TITLE
Accept compiler options when building app

### LINKED ISSUE #33: Vessel lacks a mechanism to set options for the Clojure compiler
The Clojure compiler can be controlled via the use of several compiler flags. At runtime these are stored in the dynamic var clojure.core/*compiler-options*, that we can bind defining our own settings. Please refer to [the Clojure documentation](https://clojure.org/reference/compilation#_compiler_options) for more details.

Currently, we don't have a mechanism to provide these compiler options to Vessel: it does not bind the compiler-options dynamic var and it does not have a mechanism to provide command line options to the subprocess that performs compilation.

It would be useful for us to have such a mechanism, so that we can enable features like direct linking or eliding metas.

### PR DESCRIPTION
**Pull request by miorimmax**
* **Please check if the PR fulfills these requirements**
- [x] There is an open issue describing the problem that this pr intents to solve.
    - [x] You have a descriptive commit message with a short title (first line).
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added / updated (for bug fixes / features).

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...) 
Feature

* **What is the current behavior?**
We don't have a mechanism to provide compiler options to the Clojure compiler.

* **What is the new behavior (if this is a feature change)?**
We can provide compiler options to the Clojure compiler using the `--compiler-options OPTIONS` command line argument, whose value is a map in EDN format (i.e. anything that `clojure.edn/read-string` can read).

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
This PR does not introduce breaking changes.

* **Other information**:
Closes: https://github.com/nubank/vessel/issues/33

### CODE REVIEW

**Comment by miorimmax** [comment]
Looks like the PR checks are not triggering for some reason, anything I should do?

File: src/vessel/program.clj [hunk 1]
```diff
@@ -50,6 +51,12 @@
              :desc "Directory where the application's files will be written to"
              :parse-fn io/file
              :validate cli/file-or-dir-must-exist]
+            ["-C" "--compiler-options OPTIONS"
+             :id :compiler-options
+             :desc "Options provided to the Clojure compiler, see clojure.core/*compiler-options*"
+             :default nil
+             :parse-fn edn/read-string
+             :validate [#(or (nil? %) (map? %)) "Compiler options must be a valid Clojure map"]]
```
Comment (alan-ghelardi) [inline src/vessel/program.clj]: 💅  maybe it's worth moving this sexpr to a def since you are duplicating it below.

**Review by alan-ghelardi — APPROVED** [review]
lgtm. I'll have a look why the Action didn't trigger 😕

**Comment by alan-ghelardi** [comment]
I changed the workflow to trigger on pull_request events. Could you rebase onto the main branch, please? Also, most likely, you'll need to enable actions on your fork if they aren't enabled by default. Last, if possible, add a new entry to the changelog with a brief explanation about your change. Just put a new Added section under the Unreleased one. When we trigger a release, the changelog is bumped automatically.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+## [0.2.149] - 2024-01-31
```
Comment (alan-ghelardi) [inline CHANGELOG.md]: Please, remove this title. The release script alters the changelog automatically during the release flow. Just keep the added section under the unreleased one.

**Review by alan-ghelardi — CHANGES_REQUESTED** [review]
I'm just requesting a change in the changelog to avoid breaking the release process. The feature itself is fine.

**Review by alan-ghelardi — APPROVED** [review]

**Pull request merged**